std library compile errors for uwp-windows-msvc targets

<!--
Thank you for filing a regression report! 🐛 A regression is something that changed between versions of Rust but was not supposed to.

Please provide a short summary of the regression, along with any information you feel is relevant to replicate it.
-->

Building std library (-Zbuild-std) with nightly for uwp-windows-msvc targets has been failing with compile errors since around after 02/06/22

```rust
error[E0425]: cannot find value `reparse_tag` in this scope
   --> C:\Users\runneradmin\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib\rustlib\src\rust\library\std\src\sys\windows\fs.rs:406:21
    |
406 |                     reparse_tag = attr_tag.ReparseTag;
    |                     ^^^^^^^^^^^ not found in this scope

```

I expected to see this happen:

Last successeful build on 02/06:

https://github.com/korejan/ALVR/runs/8154433510?check_suite_focus=true

Instead, this happened:

Lastest failed build:

https://github.com/korejan/ALVR/runs/8204519566?check_suite_focus=true#step:6:275

### Version it worked on

It most recently worked on: rustc 1.65.0-nightly (2e35f954a 2022-09-01)

### Version with regression

`rustc --version --verbose`:
```
1.65.0-nightly (b44197abb 2022-09-05)
```

<!--
Did the compiler crash? If so, please provide a backtrace.
-->

### Backtrace
<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

<!--
If you know when this regression occurred, please add a line like below, replacing `{channel}` with one of stable, beta, or nightly.

@rustbot modify labels: +regression-from-stable-to-{channel} -regression-untriaged
-->

--- a/library/std/src/sys/windows/fs.rs
+++ b/library/std/src/sys/windows/fs.rs
@@ -403,7 +403,7 @@ impl File {
                     mem::size_of::<c::FILE_ATTRIBUTE_TAG_INFO>().try_into().unwrap(),
                 ))?;
                 if attr_tag.FileAttributes & c::FILE_ATTRIBUTE_REPARSE_POINT != 0 {
-                    reparse_tag = attr_tag.ReparseTag;
+                    attr.reparse_tag = attr_tag.ReparseTag;
                 }
             }
             Ok(attr)

